### PR TITLE
Reorganize file structure

### PR DESCRIPTION
**Pull request by KinanBab**
1. Put all protobuf definitions in one file
2. rename all BUILD and WORKSPACE file to .bazel
3. put the p4-sample programs in a separate directory under / since they will be shared by many components
4. put bmv2 json parsing cc file, protobufs, and test files, including expected files and python script, under src/bmv2
5. Make appropriate changes to package naming, cc includes and bazel loads, and test scripts
6. Re-write and improve veth_setup.sh and update p4-samples Makefiles.

### CODE REVIEW

File: BUILD.bazel
Comment (smolkaj) [inline BUILD.bazel]: Move to the p4-samples folder.

File: src/bmv2/BUILD.bazel
Comment (smolkaj) [inline src/bmv2/BUILD.bazel]: After moving the BUILD file as suggested above, this will become `//p4-samples:port-table/table.p4` and similarly below.

**Review by smolkaj — APPROVED** [review]
LGTM. Just a small comment.

**Comment by KinanBab** [comment]
@smolkaj done. I will wait for stefan to approve as well before merging.

File: src/bmv2/bmv2.proto
Comment (stefanheule) [inline src/bmv2/bmv2.proto]: I would remove this last paragraph. It is just going to become stale/out-of-date, and it doesn't add much. We should explain the different phases in the place where we actually do it, not here.

File: src/bmv2/bmv2.proto
Comment (stefanheule) [inline src/bmv2/bmv2.proto]: Same here, I would remove lines 35-45.

**Comment by KinanBab** [comment]
Renamed src to p4_symbolic. Path and package paths now are identical. 

Closes #9.

File: p4_symbolic/bmv2/test.cc
Comment (stefanheule) [inline p4_symbolic/bmv2/test.cc]: I don't think we want these to be in any namesapce, since this is a main executable.

**Review by stefanheule — COMMENTED** [review]
Just some minor comments.

File: p4_symbolic/bmv2/test.cc
Comment (KinanBab) [reply p4_symbolic/bmv2/test.cc]: Yeah that makes sense

**Review by stefanheule — APPROVED** [review]

**Pull request merged**
